src/encoded/static/components/matrix.js

```diff
@@ -1,367 +1,589 @@
+        const type = pluralize(queryStringType.toLocaleLowerCase());
+        return (
+            <div className="matrix-general-search">
+                <p>Enter search terms to filter the {type} included in the matrix.</p>
+                <div className="general-search-entry">
+                    <i className="icon icon-search" />
+                    <div className="searchform">
+                        <TextFilter filters={context.filters} searchBase={matrixSearch} onChange={this.onChange} />
+                    </div>
+                </div>
+            </div>
+        );
     }
 }
 
-GroupMoreButton.propTypes = {
-    id: PropTypes.string, // ID for the handleClick function to know which control was clicked
-    handleClick: PropTypes.func.isRequired, // Call this parent function to handle the click in the button
-    displayText: PropTypes.string, // Text to display in the button while closed
+SearchFilter.propTypes = {
+    /** Matrix search results object */
+    context: PropTypes.object.isRequired,
 };
 
-GroupMoreButton.defaultProps = {
-    id: '',
-    displayText: '',
+SearchFilter.contextTypes = {
+    navigate: PropTypes.func,
+    location_href: PropTypes.string,
 };
 
 
-class Matrix extends React.Component {
-    static generateYGroupOpen(matrix) {
-        // Make a state for each of the Y groups (each Y group currently shows a biosample type).
-        // To do that, we have to get each of the bucket keys, which will be the keys into the
-        // object that keeps track of whether the group shows all or not. If a group has fewer than
-        // the maximum number of items to show a See More button, it doesn't get included in the
-        // state.
-        const primaryYGrouping = matrix.y.group_by[0];
-        const secondaryYGrouping = matrix.y.group_by[1];
-        const yLimit = matrix.y.limit;
-        const yGroups = matrix.y[primaryYGrouping].buckets;
-        const yGroupOpen = {};
-        yGroups.forEach((group) => {
-            if (group[secondaryYGrouping].buckets.length > yLimit) {
-                yGroupOpen[group.key] = false;
-            }
+/**
+ * Given one subcategory of matrix data (all the subcategory rows within a category), collect
+ * summary information about that data.
+ * @param {array}  subCategoryData Array of subcategory objects, each containing an array of data
+ * @param {string} columnCategoryType `subCategoryData` property that contains data array
+ *
+ * @return {object} Summary information about given matrix data:
+ *     {
+ *         {array} subCategorySums: Column sums for all subcategory rows
+ *         {number} maxSubCategoryValue: Maximum value in all cells in all subcategory rows
+ *         {number} minSubCategoryValue: Minimum value in all cells in all subcategory rows
+ *     }
+ */
+const analyzeSubCategoryData = (subCategoryData, columnCategoryType) => {
+    const subCategorySums = [];
+    let maxSubCategoryValue;
+    let minSubCategoryValue;
```

Comment:
Could we just initialize these to be 0 and skip the check for if they are undefined in line 162?

Reply:
Almost. With minSubCategoryValue set to 0, the result for that is always zero. But I just learned Javascript has a defined MAXINT referred to as `Number​.MAX_VALUE` so I initialize minSubCategoryValue to that, with maxSubCategoryValue set to 0.

Reply:
Also looking at this revealed a very subtle bug in that analyzeSubCategoryData wasn't filtering out zero values in the data when it looked for minimum values, so it always returned zero in minSubCategoryValue anyway. I changed it to filter out zero values when checking for minimums. Also, I subtract one from the minimum value before returning it (minimum value is always guaranteed to be ≥ 1 now). Without that, minimum values in the table would always be white, which I think people would complain about. By subtracting from the minimum value, any non-zero values have at least some obvious color in the table.

Reply:
Glad you caught that! I agree that the non-zero values should not be white. I wonder if it might have been more straightforward to use a value that isn't 0 for full-width rows but I can't really think of a better value.